src/fileloader.h

```diff
@@ -131,6 +134,12 @@ class FileLoader : public QObject
             {
                 g_logger.error("Error loading %s: %s", loadInfo.filePath, e.what());
             }
+
+            if (geom.get())
+                geom->disconnect();
```

Comment:
The point is, bool(geom.get()) *always* returns true since geom cannot be null at this point.  Therefore, `geom->disconnect()` should be unconditional since the if statement only serves to bloat up the code for no reason.